src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -240,13 +238,15 @@ public function theCustomerChoseShippingWithPayment(
         $this->orderShipmentFactory->processOrderShipment($order);
         $order->getShipments()->first()->setMethod($shippingMethod);
 
-        $this->orderRecalculator->recalculate($order);
+        $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply(OrderCheckoutTransitions::TRANSITION_SELECT_SHIPPING);
 
         $payment = $this->paymentFactory->createWithAmountAndCurrencyCode($order->getTotal(), $order->getCurrencyCode());
         $payment->setMethod($paymentMethod);
-
         $order->addPayment($payment);
 
+        $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply(OrderCheckoutTransitions::TRANSITION_SELECT_PAYMENT);
+        $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply(OrderCheckoutTransitions::TRANSITION_COMPLETE);
```

Comment:
Maybe we could extract this all transitions calls to private method? Would be much cleaner:
```php
private function applyTransition(OrderInterface $order, $transition)
{
    $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply($transition);
}
```
wdyt?